packages/utils/upgrade/resources/codemods/5.0.0/useRBAC-hook-import-change.code.ts

```diff
@@ -0,0 +1,84 @@
+import type { Transform, ImportDeclaration, JSCodeshift, Collection } from 'jscodeshift';
+
+const changeImportSpecifier = (
+  root: Collection,
+  j: JSCodeshift,
+  options: { methodName: string; oldDependency: string; newDependency: string }
+): void => {
+  const { methodName, oldDependency, newDependency } = options;
+
+  // Flag to check if the method was imported from the old dependency
+  let methodImportedFromOldDependency = false;
+
+  // Remove the method from the old dependency and check if it was imported
+  root
+    .find(j.ImportDeclaration)
+    .filter((path) => path.node.source.value === oldDependency)
+    .forEach((path) => {
+      const importDeclaration: ImportDeclaration = path.node;
+
+      // Check if the method is imported from the old dependency
+      const methodExistsInOldDependency = importDeclaration.specifiers?.some(
+        (specifier) =>
+          specifier.type === 'ImportSpecifier' && specifier.imported.name === methodName
+      );
+
+      if (methodExistsInOldDependency) {
+        methodImportedFromOldDependency = true;
+
+        const updatedSpecifiers = importDeclaration.specifiers?.filter(
+          (specifier) =>
+            specifier.type === 'ImportSpecifier' && specifier.imported.name !== methodName
+        );
+
+        if (updatedSpecifiers && updatedSpecifiers.length > 0) {
+          // Replace the import with the updated specifiers if there are other imports left
+          j(path).replaceWith(j.importDeclaration(updatedSpecifiers, j.literal(oldDependency)));
+        } else {
+          // Remove the entire import statement if the specified method was the only import
+          j(path).remove();
+        }
+      }
+    });
+
+  // Add new import dependency if the method was imported from the old dependency
+  if (methodImportedFromOldDependency) {
+    const dependencies = root
+      .find(j.ImportDeclaration)
+      .filter((path) => path.node.source.value === newDependency);
+
+    dependencies.forEach((path) => {
+      const importDeclaration: ImportDeclaration = path.node;
+
+      const newSpecifier = j.importSpecifier(j.identifier(methodName));
+      importDeclaration.specifiers?.push(newSpecifier);
+    });
+    // Add the new import declaration if it doesn't already exist
+    if (dependencies.length === 0) {
+      const newImportDeclaration = j.importDeclaration(
+        [j.importSpecifier(j.identifier(methodName))],
+        j.literal(newDependency)
+      );
+      root.get().node.program.body.unshift(newImportDeclaration);
```

Comment:
```suggestion
      // Find the index of the first non-import declaration
      const body = root.get().node.program.body;
      const lastImportIndex = body.findIndex(node => node.type !== 'ImportDeclaration');

      // Check if there are any import declarations
      if (lastImportIndex > -1) {
        // Insert the new import declaration just before the first non-import node
        body.splice(lastImportIndex, 0, newImportDeclaration);
      } else {
        // Check if 'use strict' exists at the beginning
        const hasUseStrict = body[0]?.type === 'ExpressionStatement' && body[0]?.expression?.value === 'use strict';

        // Add the new import after 'use strict' if it exists, otherwise at the beginning
        body.splice(hasUseStrict ? 1 : 0, 0, newImportDeclaration);
      }
```

This will add the new import directly after the import it was removed from, rather than adding it at the beginning. That will keep all the strapi imports together instead of separating them.

To see an example of the problem, start with this code:

```
import React from 'react';
import { Button } from '@strapi/design-system';
import { useRBAC, useOtherMethod } from "@strapi/helper-plugin";
```

the original codemod transforms it to:
```
import { useRBAC } from "@strapi/strapi/admin";
import React from 'react';
import { Button } from '@strapi/design-system';
import { useOtherMethod } from "@strapi/helper-plugin";
```

the updated code transforms it to:
```
import React from 'react';
import { Button } from '@strapi/design-system';
import { useOtherMethod } from "@strapi/helper-plugin";
import { useRBAC } from "@strapi/strapi/admin";
```